consumers/writers/influxdb/consumer.go

```diff
@@ -32,19 +33,76 @@ type RepoConfig struct {
 type influxRepo struct {
 	client           influxdb2.Client
 	cfg              RepoConfig
+	errCh            chan error
+	writeAPI         api.WriteAPI
 	writeAPIBlocking api.WriteAPIBlocking
 }
 
-// New returns new InfluxDB writer.
-func New(client influxdb2.Client, config RepoConfig, async bool) consumers.Consumer {
+// NewSync returns new InfluxDB writer.
+func NewSync(client influxdb2.Client, config RepoConfig) consumers.SyncConsumer {
 	return &influxRepo{
 		client:           client,
 		cfg:              config,
+		writeAPI:         nil,
 		writeAPIBlocking: client.WriteAPIBlocking(config.Org, config.Bucket),
 	}
 }
 
-func (repo *influxRepo) Consume(message interface{}) error {
+func NewAsync(client influxdb2.Client, config RepoConfig) consumers.AsyncConsumer {
+	return &influxRepo{
+		client:           client,
+		cfg:              config,
+		errCh:            make(chan error, 1),
+		writeAPI:         client.WriteAPI(config.Org, config.Bucket),
+		writeAPIBlocking: nil,
+	}
+}
+
+func (repo *influxRepo) ConsumeAsync(message interface{}) {
+	var err error
+	var pts []*write.Point
+	switch m := message.(type) {
+	case json.Messages:
+		pts, err = repo.jsonPoints(m)
+	default:
+		pts, err = repo.senmlPoints(m)
+	}
+	if err != nil {
+		repo.errCh <- err
+		return
+	}
+
+	done := make(chan bool)
+	defer close(done)
+
+	go func(done <-chan bool) {
+		for {
+			select {
+			case err := <-repo.writeAPI.Errors():
+				repo.errCh <- err
+			case <-done:
+				repo.errCh <- nil // pass nil error to the error channel
+				return
+			}
+		}
+	}(done)
+
+	for _, pt := range pts {
+		repo.writeAPI.WritePoint(pt)
+	}
+
+	repo.writeAPI.Flush()
```

Comment:
Instead of manually flushing, set a flush interval and let the client take care of it. Looks like defaults are pretty good for us, so you can just omit this flush.
(Defaults:
```go
// DefaultOptions returns Options object with default values
func DefaultOptions() *Options {
	return &Options{batchSize: 5_000, flushInterval: 1_000, precision: time.Nanosecond, useGZip: false, retryBufferLimit: 50_000, defaultTags: make(map[string]string),
		maxRetries: 5, retryInterval: 5_000, maxRetryInterval: 125_000, maxRetryTime: 180_000, exponentialBase: 2}
}
```
)

Reply:
Tests fail if I don't flush at the end. The messages don't get saved.
I tried default options with `SetFlushInterval(time.Millisecond)` to 1000, 100, 10, and 1. And all of these still cause the tests to fail.

Reply:
Tested, looks like a bug in InfluxDB client.